neonvm/runner/main.go

```diff
@@ -299,6 +301,58 @@ func createISO9660runtime(
 		return err
 	}
 
+	if swapInfo != nil && swapInfo.Shrinkable != nil && *swapInfo.Shrinkable {
+		lines := []string{
+			`#!/neonvm/bin/sh`,
+			`set -euxo pipefail`,
+			// this script may be run as root, so we should avoid potentially-malicious path
+			// injection
+			`export PATH="/neonvm/bin"`,
+			`swappart="$(blkid -L swapdisk)"`,                       // e.g. '/dev/vdd1'
+			`swapdisk="$(echo "$swappart" | grep -oE '^/dev/vd.')"`, // e.g. '/dev/vdd'
+			// disable swap. Allow it to fail if it's already disabled.
+			`swapoff $swappart || true`,
+			// if the requested size is zero, then... just exit. There's nothing we need to do.
+			`new_size="$1"`,
+			`if [ "$new_size" = '0' ]; then exit 0; fi`,
+			// Find the new end position of the partition, which is what 'parted' requires.
+			// To do this, we'll inspect the current partition.
+			//
+			// 'parted -m <device> unit s print' typically shows something like:
+			//
+			//     BYT;
+			//     /dev/vdd:20971520s:virtblk:512:512:gpt:Virtio Block Device:;
+			//     1:4096s:20969471s:20965376s:linux-swap(v1):primary:;
+			//
+			// So here, we're just running it through the standard unix commands to get the integer
+			// values of the fields we want. (note: 'unit s' means output is measured in sectors)
+			`partinfo="$(parted -m "$swapdisk" unit s print)"`,
+			// in the example above: 512
+			`sector_size="$(echo "$partinfo" | tail -n+2 | head -1 | cut -d: -f4 | grep -oE '[[:digit:]]+')"`,
+			// in the example above: 4096
+			`part_start="$(echo "$partinfo" | tail -n1 | cut -d: -f2 | grep -oE '[[:digit:]]+')"`,
```

Comment:
I think this is too complicated of a logic for bash script.

Reply:
How so? we're basically just getting the Nth fields of a couple lines of output.

Do you have a suggestion for what it should be written in instead?

see also: similar (but not the same) logic implemented in go: https://github.com/neondatabase/autoscaling/blob/eb6fee94c93f57c83c7785f6791113f9d84cc6ad/neonvm/runner/main.go#L462-L484

Reply:
Implementation changed.